### DISCUSSION TITLE
Supporting multiple hydration directives

### DISCUSSION THREAD

**Discussion by ohansemmanuel** [withastro/docs#3079] [Ideas]
Hi there! 

I haven't had the chance to check the source, but out of curiosity, does Astro support passing multiple hydration directives when hydrating a framework component? 

For example: 

``` js 

// hydrate on load or when visible, whichever is triggered first 
<ImagesComponent client:visible client:idle />
```

Some island implementations support this e.g., Eleventy's minimal [is-land](https://github.com/11ty/is-land). 

Any thoughts?

**Comment by ohansemmanuel** [comment]
Okay, I gave this a try and it's not supported. At least with the `idle` and `visible` directives together. 

Does seem like there are legitimate use cases for this though.

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
